feat: use short names for DisplayName

Dynge opened the pull request:
This PR uses regex to truncate the DisplayName to the last part.

This removes the namespace from the displayname that will be duplicated across all tests in the same namespace.

Nsidorenco commented:
Hey 👋🏻 

Great idea to shorten the names!
I think it is probably more suited to move this into the Lua code, where we could probably do the test name manipulation wrt. to the file we're displaying the test under.  With the regex solution I think we would accidentally cut test names which contains a dot in the display name.

I'm not too familiar with test adapters outside `Xunit`, but I have a hunch that other test adapters might not set the display name of test as `namespace.class.method`. I wonder how something like VS/Rider chooses the which name to display.

Dynge commented:
Hey

Yeah I am not familiar with tests suites other than xunit either. I could move it to the Lua part. I also think it is a good idea to trim the name wrt the namespace and file that the test is placed under Instead of this naive regex.

Dynge commented:
Updated the PR to instead do the shortening of DisplayName when lua reads the json data.

Also updated the tests and added a test using DisplayName = "" in the Fact attribute

Nsidorenco commented:
Really great stuff! I'll try it out on a couple of solutions and get back to you!

Nsidorenco commented:
Looks great! Thank you for this

Dynge commented:
Thank you @Nsidorenco 

If I may ask a favor; i assume you also use xunit as a testing framework and netcoredbg to debug. 

Do you experience issues running DAP with xunit.v3?

Ive had some issues (see https://github.com/Samsung/netcoredbg/issues/188 - reproducible repo included) but I am confused if its a consistent problem with netcoredbg and xunit.v3 or if its more flaky.

I cannot hit breakpoints if I switch to xunit.v3, can you?

Also - entirely fair if you do not have time or wish to embark on this tangent 😅
